runbooks/how-to-do-a-deployment.md

```diff
@@ -0,0 +1,52 @@
+
+6. Ensure no other jobs are currently building or queued up to build; it's also a good practice to check if another developer is performing work in the ```Test``` environment. This can be done by checking the ```Slack``` channel ```#bfd-builds```. If the ```Test``` environment is not clear, then one should either postpone the build, or contact the developer to see if they can relinqush their claim to the ```Test``` environment.
+
+7. While not required, it is convenient to notify BFD developers of the intent to perform a deployment; this can be done by posting a message to the ```#bfd-builds``` Slack channel.
+    - post message: ```/dibs on test``` to stake claim to the test enviornemnt.
+    - post message: ```/dibs off test``` to release claim to the test enviornemnt.
+
+8. From the ```Stage View``` for ```branch master```, selecting ```Build with Parameters``` link on left-side displays a set of parameters that can be applied to the pipeline build:
+    - `deploy_prod_from_non_master` : if enabled, deploy to prod-like envs for builds of this project's non-master branches.
+    - `deploy_prod_skip_confirm` : if enabled, skips prompting for confirmation before deploying to most prod-like envs.
+    - `build_platinum` : if enabled, build/update the _platinum_ base AMI; while no hard-fast rule for when to deploy an updated AMI, it is generally done about once/week. An quick way to ascertain the last deployment, is to look at the ```Build Platinum AMI``` column in the ```Stage``` view; any builds of the AMI will show there.
+    - `use_latest_images` : if enabled, defer to latest available AMIs; skips `App` and `App Image` Stages.
+    - `verbose_mvn_logging` : if enabled, `mvn` will produce verbose logs.
+    - `skip_migrator_deployment` : if enabled, AND building a non-trunk or non-master branch, then do not deploy the migrator to the `test` environment.
+    - `server_regression_image_override` : text input that can be used to identify/override the `Docker` image tag used when deploying the server-regression lambda.
+
+9. Activate the pipeline processing using the ```Build``` button, which will include any parameters selected above.
+
+10. When build commences, the ```Stage View``` is displayed showing the now activated current pipeline processing stage(s). If you need to see more detail of the pipeline build, then click the current build number (under ```Build History```). Some useful actions that provide more detailed information and control:
+    - ```Console Output``` provides a detailed view of the console output log.
+    - ```Pipeline Steps``` provides a dynamic hierarchial view of the pipeline stages.
+    - ```Pause/resume``` allows you to stop (and restart) the pipeline processing.
+
+11. When the build pipeline finishes the deployment to ```Test```, the pipeline may pause requesting a ```Manual Approval``` to proceed on to ```prod-sbx, prod```. This gives the operator a chance to abort further processing if needed.
```

Comment:
The environment-specific `Server` stages run the regression suite at the conclusion of each stage–this has historically been a top contributor to deployment _failures_ in the past 6 months (to be fair, Mitch has rooted the false positives out by now). And on path-to-prod, `master` branch-based deployments, we should expect this to fail when there is a serious performance problem. I'm not sure if this is the place to document it, but... maybe it might be helpful to note:
* for deployments based on `master`, we visualize historical regression results in AWS QuickSight for `test`, `prod-sbx`, and `prod`
* for all deployments, Jenkins logs point operator to the raw CloudWatch-stored regression suite results

Maybe this is a need that could be better answered by general deployment operations documentation?